menas/src/main/scala/za/co/absa/enceladus/menas/controllers/SchemaController.scala

```diff
@@ -48,6 +51,35 @@ class SchemaController @Autowired()(
 
   import scala.concurrent.ExecutionContext.Implicits.global
 
+  @PostMapping(Array("/remote"))
+  @ResponseStatus(HttpStatus.CREATED)
+  def handleRemoteFile(@AuthenticationPrincipal principal: UserDetails,
+                       @RequestParam remoteUrl: String,
+                       @RequestParam version: Int,
+                       @RequestParam name: String,
+                       @RequestParam format: Optional[String]): CompletableFuture[Option[Schema]] = {
+
+    val url = new URL(remoteUrl)
```

Comment:
If there was a reason for it, the URL retrieval could also be wrapped in some kind of catch-try block in case we are not happy with the default stack trace. Unsure if we wanted a specific Exception for this case or an existing one to be used here, I did not introduce such exception handling. Could be done if we wanted to.

Reply:
There should be something, line 62 throws an error in case the host doesn't exist and line 63 if the host is something like www.google.com. Both propagate to the user as 500 internal server errors. Could we tell user something better? Or give them a status code that it is their fault?

Reply:
Agree. If by creating our own exception wrapper the user will get a better error message so they know how to fix it, it is worth doing it.

Reply:
Done, now it returns code `400` and we determine by the error type on the front-end which error message to show to the user.